### PR TITLE
Adds color conversion to SRGB for images with CMYK palette

### PR DESCRIPTION
**Pull request by krisdante**
### Description (*)

Adds color conversion to SRGB for images with CMYK and Unrecognizable color palette

For ImageMagic, when the original image was in CMYK palette, the final image after
resize was kind of "in negative". I decided to force conversion to RGB, but when
converting to RGB from CMYK with IMagixk - colors were not natural.
So I picked SRGB (I found this hint in StackOverflow)

### Fixed Issues (if relevant)

1. magento/magento2#22375  : CMYK images are rendered with inverted colors

### Manual testing scenarios (*)

See the issue for a detailed path to reproduce / test

### Questions or comments

It might be that "normal" RGB images will get converted to SRGB in some cases. I tested a few RGB images and I do not see the quality problem.

I fixed a few lines in this file were phpcs was complaining. 

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by krisdante** [comment]
Those Exceptions (Code style failing) were there before. I think migrating that is a separate topic and I will do that with another pull request.

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests CE, Functional Tests B2B, Functional Tests EE

File: lib/internal/Magento/Framework/Image/Adapter/ImageMagick.php [hunk 1]
```diff
@@ -512,4 +513,20 @@ protected function _getImagickPixelObject($color = null)
     {
         return new \ImagickPixel($color);
     }
+
+    /**
+     * Converts colorspace to RGB when required
+     */
+    protected function ensureColorspace()
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/Image/Adapter/ImageMagick.php:513]: Please make this method private and add return type, like this.
```suggestion
    private function ensureColorspace(): void
```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @krisdante,
Your change looks good to me. Could you cover your changes with unit test?

**Comment by ihor-sviziev** [comment]
Hi @krisdante,
Will you be able to update your PR?

**Comment by krisdante** [comment]
I need to think about it. The tricky part with unit testing that is that you cannot check colours programatically. They will be fine. They are just displayed brokenly to a human eye when wrong colour space conversion will happen. 
The test here was to upload the image attached to this bug, and check if it is looking good in Magento. 
It is really hard to program this as integration test.

**Comment by krisdante** [comment]
Also merging this before https://github.com/magento/magento2/pull/26036 would be nice.

File: lib/internal/Magento/Framework/Image/Adapter/ImageMagick.php [hunk 2]
```diff
@@ -512,4 +513,20 @@ protected function _getImagickPixelObject($color = null)
...
+    {
+        $colorspace = $this->_imageHandler->getColorspace();
+        switch ($colorspace) {
+            case \Imagick::COLORSPACE_RGB:
+                return;
+            case \Imagick::COLORSPACE_CMYK:
+            case \Imagick::COLORSPACE_UNDEFINED:
+                $this->_imageHandler->transformImageColorspace(\Imagick::COLORSPACE_SRGB);
+
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/Image/Adapter/ImageMagick.php:520]:
```suggestion
                break;
```

File: lib/internal/Magento/Framework/Image/Adapter/ImageMagick.php [hunk 3]
```diff
@@ -512,4 +513,20 @@ protected function _getImagickPixelObject($color = null)
...
+    /**
+     * Converts colorspace to RGB when required
+     */
+    protected function ensureColorspace()
+    {
+        $colorspace = $this->_imageHandler->getColorspace();
+        switch ($colorspace) {
+            case \Imagick::COLORSPACE_RGB:
+                return;
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/Image/Adapter/ImageMagick.php]: I think we could just remove this case.

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
> I need to think about it. The tricky part with unit testing that is that you cannot check colours programatically. They will be fine. They are just displayed brokenly to a human eye when wrong colour space conversion will happen.
The test here was to upload the image attached to this bug, and check if it is looking good in Magento.
It is really hard to program this as integration test.


@krisdante yeah, that's good point for sure, but I didn't see any integration tests for image magic, but this would be really good thing if you could write it!
By unit test we could just check that your code was executed and in case if you'll write integration - it will be enough. If you'll have any issues with integration tests - unit test we still could write without any issues.

Also I wrote few comments to existing functionality, please check them

Could you update your PR accordingly?

**Comment by ihor-sviziev** [comment]
Hi @krisdante,
Will you be able to update your PR?

**Comment by ihor-sviziev** [comment]
@krisdante  I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**